appcues/src/main/java/com/appcues/ui/presentation/ViewPresenter.kt

```diff
@@ -129,7 +129,7 @@ internal abstract class ViewPresenter(
 
     fun remove() {
         // ensure all this is running on the main thread
-        Handler(Looper.getMainLooper()).post {
+        postIfNeeded {
```

Comment:
seems like we almost always do not need to post to the main thread here, and it would be complex to make this suspending, so I updated to this check for the main thread and only post if needed.  Tests are not stable without this, and they are already on the main thread when the dismissal occurs.

Reply:
Thats right, this was to help with some inconsistencies when we had multiple activities (particularly transparent)